framework/src/mesh/MooseMesh.C

```diff
@@ -355,7 +355,14 @@ MooseMesh::prepare(const bool force_mesh_prepare)
     // For whatever reason we do not want to allow renumbering here nor ever in the future?
     getMesh().allow_renumbering(false);
 
-  if (!_mesh->is_prepared() || force_mesh_prepare)
+  if (mesh_to_clone)
+  {
+    mooseAssert(mesh_to_clone->is_prepared(),
+                "The mesh we wish to clone from must already be prepared");
+    _mesh = mesh_to_clone->clone();
+    _moose_mesh_prepared = false;
```

Comment:
is this "currently prepared" status or "I just prepared it status" ?

seems like the clone is prepared

Reply:
The clone is prepared (both the MeshBase and the MooseMesh presumably from which it came) but "we" are not, e.g. "our" MooseMesh is not prepared

Reply:
ah I get it now

Reply:
This is kind of horrifying.  clone() isn't exactly cheap.  It's also going to regenerate a lot of EquationSystems-specific indexing ... which I guess isn't a problem where where we expect that indexing should be the same regardless.

I guess it's necessary in cases where adaptivity goes out of sync?